File: VB/CustomBandedGridView/CustomGrid.vb
Imports DevExpress.Utils
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.BandedGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.Handler
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Views.Printing
Imports DevExpress.XtraPrinting
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports System.Windows.Forms

Namespace dxExample
    <ToolboxItemAttribute(True)> _
    Public Class CustomGridControl
        Inherits GridControl

        Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
            MyBase.RegisterAvailableViewsCore(collection)
            collection.Add(New MyGridViewInfoRegistrator())
            collection.Add(New MyBandedGridViewInfoRegistrator())
        End Sub

    End Class
    Public Class MyGridViewInfoRegistrator
        Inherits GridInfoRegistrator

        Public Overrides ReadOnly Property ViewName() As String
            Get
                Return CustomView.SViewName
            End Get
        End Property
        Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
            Return New CustomBandedGridView(TryCast(grid, GridControl))
        End Function
    End Class
    Public Class MyBandedGridViewInfoRegistrator
        Inherits BandedGridInfoRegistrator

        Public Overrides ReadOnly Property ViewName() As String
            Get
                Return CustomBandedGridView.SViewName
            End Get
        End Property
        Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
            Return New CustomBandedGridView(TryCast(grid, GridControl))
        End Function
    End Class

End Namespace

File: CS/CustomBandedGridView/CustomGrid.cs
using DevExpress.Utils;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.BandedGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.Handler;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Views.Printing;
using DevExpress.XtraPrinting;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Windows.Forms;

namespace dxExample {
    [ToolboxItemAttribute(true)]
    public class CustomGridControl : GridControl {

        protected override void RegisterAvailableViewsCore(InfoCollection collection) {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new MyGridViewInfoRegistrator());
            collection.Add(new MyBandedGridViewInfoRegistrator());
        }

    }
    public class MyGridViewInfoRegistrator : GridInfoRegistrator {
        public override string ViewName { get { return CustomView.SViewName; } }
        public override BaseView CreateView(GridControl grid) {
            return new CustomBandedGridView(grid as GridControl);
        }
    }
    public class MyBandedGridViewInfoRegistrator : BandedGridInfoRegistrator {
        public override string ViewName { get { return CustomBandedGridView.SViewName; } }
        public override BaseView CreateView(GridControl grid) {
            return new CustomBandedGridView(grid as GridControl);
        }
    }
    
}
